lib/room.cpp

```diff
@@ -1487,7 +1487,13 @@ RoomEventPtr Room::decryptMessage(const EncryptedEvent& encryptedEvent)
         // qCWarning(E2EE) << "Encrypted message is empty";
         return {};
     }
-    return encryptedEvent.createDecrypted(decrypted);
+    auto decryptedEvent = encryptedEvent.createDecrypted(decrypted);
+    if (decryptedEvent->roomId() == id()) {
+        return decryptedEvent;
+    } else {
+        qCWarning(E2EE) << "Decrypted event" << encryptedEvent.id() << "not for this room; discarding.";
+        return nullptr;
```

Comment:
```suggestion
        return {};
```
Just in case `decryptMessage` return value becomes an `Omittable` ;)